src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -1,25 +1,57 @@
-import { HTTPError } from "../../../models/HTTPError";
+import moment from "moment";
+import { Service } from "../../../models/injector/ServiceDecorator";
 import * as enums from "../../../assets/Enums";
-import { ITestResult } from "../../../models/ITestResult";
-import { TestResultsDAO } from "../../../models/TestResultsDAO";
 import { DateProvider } from "./DateProvider";
 import { ITestDataProvider } from "./ITestDataProvider";
-import { Service } from "../../../models/injector/ServiceDecorator";
+import * as models from "../../../models";
+import * as utils from "../../../utils";
 
 @Service()
 export class TestDataProvider implements ITestDataProvider {
-  public testResultsDAO: TestResultsDAO | undefined;
+  public testResultsDAO: models.TestResultsDAO | undefined;
+
+  /**
+   * To fetch test results by systemNumber
+   * @param filters filters used to search the database
+   */
+  public async getTestResultBySystemNumber(filters: models.ITestResultFilters): Promise<models.ITestResult[]> {
+    try {
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
+      const result = await this.testResultsDAO.getBySystemNumber(filters.systemNumber);
+      const response: models.ITestResultData = { Count: result.Count, Items: result.Items };
+      const testResults: models.ITestResult[] = utils.ValidationUtil.getTestResultItems(response);
+      return TestDataProvider.applyTestResultsFilters(testResults, filters);
+    } catch (error) {
+       console.error("TestDataProvider.getTestResultBySystemNumber: error-> ", error);
+       throw error;
+    }
+  }
 
-  public async getTestHistory(systemNumber: string): Promise<ITestResult[]> {
+  public async getTestResultByTesterStaffId(filters: models.ITestResultFilters): Promise<models.ITestResult[]> {
+    try {
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
+      const result = await this.testResultsDAO.getByTesterStaffId(filters.testerStaffId);
+      if (result && !result.length) {
+        return result;
+      }
+      return TestDataProvider.applyTestResultsFilters(result, filters);
+    } catch (error) {
+      console.error("TestDataProvider.getTestResultBySystemNumber: error-> ", error);
+      throw error;
+   }
+  }
+
+
+  public async getTestHistory(systemNumber: string): Promise<models.ITestResult[]> {
     const fromDateTime = new Date(1970, 1, 1);
     const toDateTime = new Date();
-    let result: ITestResult[] = [];
+    let result: models.ITestResult[] = [];
     try {
-      this.testResultsDAO = this.testResultsDAO as TestResultsDAO;
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
       const data = await this.testResultsDAO.getBySystemNumber(systemNumber);
       console.log(`getTestHistory: Data Count -> ${data?.Count}`);
       if (data?.Count) {
```

Comment:
We can flatten this out with a guard that returns if there are no results

Reply:
done